src/v/cluster/metadata_dissemination_handler.cc

```diff
@@ -49,29 +51,28 @@ metadata_dissemination_handler::update_leadership_v2(
 
 ss::future<update_leadership_reply>
 metadata_dissemination_handler::do_update_leadership(
-  fragmented_vector<ntp_leader_revision> leaders) {
+  chunked_vector<ntp_leader_revision> leaders) {
     vlog(clusterlog.trace, "Received a metadata update");
-    co_await ss::parallel_for_each(
-      boost::irange<ss::shard_id>(0, ss::smp::count),
-      [this, leaders = std::move(leaders)](ss::shard_id shard) {
```

Comment:
Is there a specific reason you switched to `ss::do_with`? 

Just doing `&leaders` should do the same no? The coroutine will keep it alive until ss::parallel_for_each returns?

Reply:
This part is called very often, i dropped coroutines to prevent allocations, in this case the coroutine do not help much with readability

Reply:
why not use `_leaders.invoke_on_all(...)`?

Reply:
> This part is called very often, i dropped coroutines to prevent allocations, in this case the coroutine do not help much with readability

I really wouldn't overthink this in general. `do_with` will need an extra alloc as well to store the state.